### PR TITLE
run, create: add --passwd-entry

### PR DESCRIPTION
**Pull request by giuseppe**
It allows to customize the entry that is written to the `/etc/passwd`
file when --passwd is used.

Closes: https://github.com/containers/podman/issues/13185

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM
@containers/podman-maintainers PTAL

File: test/e2e/run_passwd_test.go [hunk 1]
```diff
@@ -137,4 +137,16 @@ USER 1000`, ALPINE)
 		Expect(run).Should(Exit(0))
 		Expect(run.OutputToString()).NotTo((ContainSubstring("1234:1234")))
 	})
+
+	It("podman run --passwd-entry flag", func() {
+		run := podmanTest.Podman([]string{"run", "--user", "1234:1234", "--passwd-entry=FOO", ALPINE, "cat", "/etc/passwd"})
+		run.WaitWithDefaultTimeout()
+		Expect(run).Should(Exit(0))
+		Expect(run.OutputToString()).To(ContainSubstring("FOO"))
```
Comment (TomSweeneyRedHat) [inline test/e2e/run_passwd_test.go:145]: Can we test that the entry is in /etc/passwd too?

File: docs/source/markdown/podman-run.1.md [hunk 2]
```diff
@@ -781,6 +781,12 @@ Override the OS, defaults to hosts, of the image to be pulled. For example, `win
 Allow Podman to add entries to /etc/passwd and /etc/group when used in conjunction with the --user option.
 This is used to override the Podman provided user setup in favor of entrypoint configurations such as libnss-extrausers.
 
+#### **--passwd-entry**=*ENTRY*
+
+Customize the entry that is written to `/etc/passwd` when `--passwd` is used.
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-run.1.md:786]: I think we might want to clarify which /etc/passwd:
```suggestion
Customize the entry that is written to the `/etc/passwd` file within the container when `--passwd` is used.
```
If you take this or tweak it, please do the same in the other man page.

File: test/e2e/run_passwd_test.go (hunk 1 above)
Comment (giuseppe) [reply test/e2e/run_passwd_test.go:145]: we are testing that.  The container runs `cat /etc/passwd` and we expect to find `FOO`, not the UID or GID.

I'll make the test clearer and add a comment.

**Comment by giuseppe** [comment]
suggestions amended

**Review by flouthoc — APPROVED** [review]
LGTM

File: pkg/specgenutil/specgen.go [hunk 3]
```diff
@@ -832,6 +832,9 @@ func FillOutSpecGen(s *specgen.SpecGenerator, c *entities.ContainerCreateOptions
 	if s.Passwd == nil {
 		s.Passwd = &t
 	}
+
+	s.PasswdEntry = c.PasswdEntry
```
[the remark this answers is not in the document]
Comment (giuseppe) [reply pkg/specgenutil/specgen.go]: fixed and pushed a new version

**Pull request merged**